src/VisualStudio/Core/Test.Next/Services/SolutionAssetCacheTests.cs

```diff
@@ -57,5 +61,29 @@ public async Task TestCleanup()
             // it should not reach here
             Assert.True(false, "asset not cleaned up");
         }
+
+        [Fact]
+        public async Task TestSolutionKeepsAssetPinned()
```

Comment:
this follow the logic of the above test, but this time ensuring that we do *not* purge a pinned item.